pallets/liquidity-proxy/src/lib.rs

```diff
@@ -1053,6 +1122,164 @@ impl<T: Config> Pallet<T> {
         Ok(sources_set)
     }
 
+    /// Calculates the max potential weight of inner_exchange
+    ///
+    /// This function should cover the current code map and all possible calls of some functions that can take a weight.
+    /// The current code map:
+    ///
+    /// inner_exchange()
+    ///     new_trivial()
+    ///     exchange_sequence()
+    ///         select_best_path()
+    ///             quote_pairs_with_flexible_amount() - call M times, where M is a count of paths
+    ///                 quote_single()
+    ///                     list_liquidity_sources()
+    ///                     quote()
+    ///                     smart_split()
+    ///                         quote()
+    ///                         quote()
+    ///                         check_rewards()
+    ///                         quote()
+    ///                         check_rewards()
+    ///         calculate_input_amount() - call only for SwapAmount::WithDesiredOutput
+    ///             quote_single()
+    ///                 list_liquidity_sources()
+    ///                 quote()
+    ///                 smart_split()
+    ///                     quote()
+    ///                     quote()
+    ///                     check_rewards()
+    ///                     quote()
+    ///                     check_rewards()
+    ///         exchange_sequence_with_input_amount()
+    ///             exchange_single()
+    ///                 quote_single()
+    ///                     list_liquidity_sources()
+    ///                     quote()
+    ///                     smart_split()
+    ///                         quote()
+    ///                         quote()
+    ///                         check_rewards()
+    ///                         quote()
+    ///                         check_rewards()
+    ///                 exchange() - call N times, where N is a count of assets in the path
+    ///
+    /// Dev NOTE: if you change the logic of liquidity proxy, please sustain inner_exchange_weight() and code map above.
+    pub fn inner_exchange_weight(
+        dex_id: &T::DEXId,
+        input: &T::AssetId,
+        output: &T::AssetId,
+        swap_variant: SwapVariant,
+    ) -> Weight {
+        let dex_info = dex_manager::Pallet::<T>::get_dex_info(dex_id).unwrap();
```

Comment:
Are there any cases where these unwrap calls may cause panic?

Reply:
The same ways when it can panic inside the code.
It seems that in this case there is no difference when the panic will be triggered, sooner or later.